js/document.js

```diff
@@ -143,52 +139,28 @@ class Document extends Window {
 	createElement(name) {
 		
 		name = name.toLowerCase();
-		
-		if (name.indexOf('canvas') >= 0) {
-			
-			if ( ! this._isCanvasRequested ) {
-				this._isCanvasRequested = true;
-				return this;
-			}
-			
-			const that = this;
-			
-			return {
-				_ctx   : null,
-				width  : this.width,
-				height : this.height,
-				
-				getContext(kind) {
-					this._ctx = that.getContext(kind);
-					return this._ctx;
-				},
-				
-				get data() {
-					return this._ctx && this._ctx.data;
-				},
-				
-				onkeydown    : () => {},
-				onkeyup      : () => {},
-				onmousedown  : () => {},
-				onmouseup    : () => {},
-				onwheel      : () => {},
-				onmousewheel : () => {},
-				onresize     : () => {},
-				
-				dispatchEvent       : () => {},
-				addEventListener    : () => {},
-				removeEventListener : () => {},
```

Comment:
These should be inherited by using the Document instance as the canvas's prototype

Reply:
But this one is no real canvas, this one is a sub-canvas. The problem here is that canvas (Document) equals a new platform window, and it is not intended.

For example three.js library uses an instance of canvas to stretch non-pow2 textures before upload. If we allowed touching THIS canvas it would spoil it's dimensions, and if we spawned a new Document it would open a new window. For that matter there was a rule: first canvas element is THIS, all other canvas elements are mocks.

If you are still sure it should work, first check against a [non-pow2 texture example](https://github.com/node-3d/3d-core-raub/blob/master/examples/three/nonpow2.js) with three.js. May be there were some more issues, but I can remember only this one.